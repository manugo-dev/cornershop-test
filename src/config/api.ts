import { create } from 'apisauce';

export enum StatusCodes {
  UNAUTHORIZED = 401
}

export const BASE_URL =
  process.env.NODE_ENV === 'test' ? 'http://app' : process.env.REACT_APP_API_BASE_URL || '';

const api = create({
  baseURL: BASE_URL,
  headers: {
    // eslint-disable-next-line @typescript-eslint/naming-convention
    'Content-Type': 'application/json'
  }
});

export default api;
