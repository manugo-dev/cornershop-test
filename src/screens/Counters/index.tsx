import { useState } from 'react';
import cn from 'classnames';

import useModal from 'hooks/useModal';
import CreateCounter from 'screens/CreateCounter';

import Search from './components/Search';
import CounterList from './components/CounterList';
import styles from './styles.module.scss';
import CounterActions from './components/CounterActions';

function Counters() {
  const { isVisible: isModalVisible, hideModal, showModal } = useModal();
  const [search, setSearch] = useState<string | undefined>();
  const [searchActive, setSearchActive] = useState(false);

  const onSearchChange = (value?: string) => {
    setSearch(value);
    if (searchActive) {
      setSearchActive(false);
    }
  };

  const onSearchFocus = () => {
    if (!search || search.length === 0) setSearchActive(true);
  };

  const onSearchBlur = () => {
    setSearchActive(false);
  };

  return (
    <>
      <header className={styles.header}>
        <Search
          className={styles.search}
          onChange={onSearchChange}
          onFocus={onSearchFocus}
          onBlur={onSearchBlur}
        />
      </header>
      <section className={cn('column', styles.content, { [styles.overflow]: searchActive })}>
        <CounterList search={search} />
        <CreateCounter.Modal isModalVisible={isModalVisible} hideModal={hideModal} />
      </section>
      <footer className={cn(styles.footer, { [styles.overflow]: searchActive })}>
        <CounterActions className={styles.actions} disabled={searchActive} openCreationModal={showModal} />
      </footer>
    </>
  );
}

export default Counters;
