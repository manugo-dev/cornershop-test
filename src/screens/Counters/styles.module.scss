.header {
  height: 60px;
  position: absolute;
  top: 0;
  width: 100%;
}

.content {
  margin-top: 60px;
}
