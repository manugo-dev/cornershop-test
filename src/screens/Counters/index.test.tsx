import { screen, render, waitFor } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { Provider } from 'react-redux';
import generateStore from 'redux/store';

import { BASE_URL } from 'config/api';

import Counters from './index';

const countersMock = [
  { id: 'abc123', title: 'List example 1', count: 0 },
  { id: 'abc124', title: 'List example 2', count: 2 }
];

const server = setupServer(
  rest.get(`${BASE_URL}/counter*`, (req, res, ctx) => res(ctx.status(200), ctx.json(countersMock)))
);

describe('#Counters', () => {
  const store = generateStore();

  beforeAll(() => server.listen());
  afterEach(() => server.resetHandlers());
  afterAll(() => server.close());

  it('It must render without errors', async () => {
    const { container } = render(
      <Provider store={store}>
        <Counters />
      </Provider>
    );
    await waitFor(() => expect(screen.getByText(countersMock[0].title)).toBeInTheDocument());
    expect(container).toMatchSnapshot();
  });
});
