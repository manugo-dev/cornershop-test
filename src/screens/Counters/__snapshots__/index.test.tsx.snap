// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Counters It must render without errors 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="row middle m-auto bar search"
    >
      <div
        class="row middle wrap wrapper input search"
      >
        <input
          class="input with-icon"
          placeholder="searchCounters"
          value=""
        />
        <span
          class="icon"
        >
          <svg
            class="column middle"
            fill="none"
            height="17"
            viewBox="0 0 17 17"
            width="17"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.93848 13.8926C8.33594 13.8926 9.64551 13.4707 10.7354 12.75L14.585 16.6084C14.8398 16.8545 15.165 16.9775 15.5166 16.9775C16.2461 16.9775 16.7822 16.4062 16.7822 15.6855C16.7822 15.3516 16.668 15.0264 16.4219 14.7803L12.5986 10.9395C13.3896 9.82324 13.8555 8.45215 13.8555 6.97559C13.8555 3.16992 10.7441 0.0585938 6.93848 0.0585938C3.13281 0.0585938 0.0214844 3.16992 0.0214844 6.97559C0.0214844 10.7812 3.13281 13.8926 6.93848 13.8926ZM6.93848 12.0469C4.15234 12.0469 1.86719 9.76172 1.86719 6.97559C1.86719 4.19824 4.15234 1.9043 6.93848 1.9043C9.72461 1.9043 12.0098 4.19824 12.0098 6.97559C12.0098 9.76172 9.72461 12.0469 6.93848 12.0469Z"
              fill="var(--grey)"
            />
          </svg>
        </span>
      </div>
    </div>
  </header>
  <section
    class="column content"
  >
    <div
      class="row m-bottom-4 counters"
    >
      <span
        class="text bold m-right-1"
      >
        itemCount {"count":2}
      </span>
      <span
        class="text m-right-1"
      >
        refreshCount {"count":1}
      </span>
      <button
        aria-label="refresh"
        class="button clean"
        type="button"
      >
        <span>
          <svg
            class="m-bottom-1"
            fill="none"
            height="18"
            viewBox="0 0 15 18"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.26562 9.09375C7.51562 9.09375 7.72656 9.01562 7.875 8.85156L11.0781 5.61719C11.2734 5.42969 11.3594 5.21094 11.3594 4.96094C11.3594 4.72656 11.2656 4.49219 11.0781 4.3125L7.88281 1.05469C7.71875 0.882812 7.51562 0.796875 7.26562 0.796875C6.78125 0.796875 6.40625 1.19531 6.40625 1.6875C6.40625 1.92969 6.49219 2.11719 6.64844 2.28125L8.34375 3.94531C8.03125 3.90625 7.71094 3.88281 7.39844 3.88281C3.50781 3.88281 0.398438 6.99219 0.398438 10.9062C0.398438 14.8125 3.54688 17.9609 7.44531 17.9609C11.3516 17.9609 14.4922 14.8125 14.4922 10.9062C14.4922 10.3594 14.1172 9.98438 13.5859 9.98438C13.0703 9.98438 12.7109 10.3594 12.7109 10.9062C12.7109 13.8203 10.3672 16.1719 7.44531 16.1719C4.53125 16.1719 2.1875 13.8203 2.1875 10.9062C2.1875 7.94531 4.49219 5.60938 7.39844 5.60938C7.84375 5.60938 8.25 5.64062 8.60938 5.69531L6.64844 7.63281C6.49219 7.79688 6.40625 7.98438 6.40625 8.21875C6.40625 8.71094 6.78125 9.09375 7.26562 9.09375Z"
              fill="var(--black)"
            />
          </svg>
        </span>
      </button>
    </div>
    <ul
      class="list"
    >
      <li
        class="row middle full-width m-bottom-3 element"
      >
        <button
          aria-label="title"
          class="button clean m-right-5 title"
          type="button"
        >
          <span>
            List example 1
          </span>
        </button>
        <div
          class="row middle m-left-auto actions"
        >
          <button
            aria-label="decrement"
            class="button clean m-right-2 button"
            disabled=""
            type="button"
          >
            <span>
              <svg
                fill="none"
                height="2"
                viewBox="0 0 14 2"
                width="14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="var(--grey)"
                  height="2"
                  rx="1"
                  width="14"
                />
              </svg>
            </span>
          </button>
          <span
            aria-label="count"
            class="text m-right-2 grey counter"
          >
            0
          </span>
          <button
            aria-label="increment"
            class="button clean m-right-2 button"
            type="button"
          >
            <span>
              <svg
                fill="none"
                height="14"
                viewBox="0 0 14 14"
                width="14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 7C0 6.44772 0.447715 6 1 6H13C13.5523 6 14 6.44772 14 7C14 7.55228 13.5523 8 13 8H1C0.447716 8 0 7.55228 0 7Z"
                  fill="var(--app-tint)"
                />
                <path
                  d="M7 14C6.44772 14 6 13.5523 6 13L6 1C6 0.447716 6.44771 2.87279e-08 7 0C7.55228 -2.87279e-08 8 0.447715 8 1L8 13C8 13.5523 7.55228 14 7 14Z"
                  fill="var(--app-tint)"
                />
              </svg>
            </span>
          </button>
        </div>
      </li>
      <li
        class="row middle full-width m-bottom-3 element"
      >
        <button
          aria-label="title"
          class="button clean m-right-5 title"
          type="button"
        >
          <span>
            List example 2
          </span>
        </button>
        <div
          class="row middle m-left-auto actions"
        >
          <button
            aria-label="decrement"
            class="button clean m-right-2 button"
            type="button"
          >
            <span>
              <svg
                fill="none"
                height="2"
                viewBox="0 0 14 2"
                width="14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="var(--app-tint)"
                  height="2"
                  rx="1"
                  width="14"
                />
              </svg>
            </span>
          </button>
          <span
            aria-label="count"
            class="text m-right-2 counter"
          >
            2
          </span>
          <button
            aria-label="increment"
            class="button clean m-right-2 button"
            type="button"
          >
            <span>
              <svg
                fill="none"
                height="14"
                viewBox="0 0 14 14"
                width="14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 7C0 6.44772 0.447715 6 1 6H13C13.5523 6 14 6.44772 14 7C14 7.55228 13.5523 8 13 8H1C0.447716 8 0 7.55228 0 7Z"
                  fill="var(--app-tint)"
                />
                <path
                  d="M7 14C6.44772 14 6 13.5523 6 13L6 1C6 0.447716 6.44771 2.87279e-08 7 0C7.55228 -2.87279e-08 8 0.447715 8 1L8 13C8 13.5523 7.55228 14 7 14Z"
                  fill="var(--app-tint)"
                />
              </svg>
            </span>
          </button>
        </div>
      </li>
    </ul>
  </section>
  <footer
    class="footer"
  >
    <div
      class="row middle actions actions"
    >
      <button
        class="button m-left-auto"
        type="button"
      >
        <span>
          <svg
            fill="none"
            height="20"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 10C0 9.21102 0.639593 8.57143 1.42857 8.57143H18.5714C19.3604 8.57143 20 9.21102 20 10C20 10.789 19.3604 11.4286 18.5714 11.4286H1.42857C0.639594 11.4286 0 10.789 0 10Z"
              fill="var(--white)"
            />
            <path
              d="M10 20C9.21102 20 8.57143 19.3604 8.57143 18.5714V1.42857C8.57143 0.639594 9.21102 4.10399e-08 10 0C10.789 -4.10399e-08 11.4286 0.639593 11.4286 1.42857L11.4286 18.5714C11.4286 19.3604 10.789 20 10 20Z"
              fill="var(--white)"
            />
          </svg>
        </span>
      </button>
    </div>
  </footer>
</div>
`;
